scripts/waterQualityScript.py

```diff
@@ -0,0 +1,204 @@
+                df = df.sort_values(by='ActivityStartDate', ascending=False)
+                city.totalDissolvedSolids = df[(df.CharacteristicName == 'Total dissolved solids') & (
+                    df.MeasureUnitCode.str.contains("mg/l"))].iloc[0].ResultMeasureValue
+                city.specificConductance = df[(df.CharacteristicName == 'Specific conductance') & (
+                    df.MeasureUnitCode.str.contains("uS/cm"))].iloc[0].ResultMeasureValue
+                city.pH = df[(df.CharacteristicName == 'pH') & (
+                    df.MeasureUnitCode.str.contains("std units"))].iloc[0].ResultMeasureValue
+                os.remove(directory + "/" + file)
+        return True
+    # if missing any or all data, expand the radius and time scale allowed for data
+    except:
+        if(not (within + 5 > 20)):
+            for file in files:
+                if(not file.startswith(".")):
+                    os.remove(directory + "/" + file)
+            cities = [city]
+            downloadFiles(cities, within + 5, startDate -
+                          timedelta(days=365), endDate, True, driver)
+            return False
+        else:
+            return True
+
+
+def downloadFiles(cities, within, startDate, endDate, replace, driver):
+    # downloading file and giving commands to read and upload it
+    directory = os.getcwd() + "/downloads"
+    for city in cities:
+        startTime = datetime.now()
+        count = citiesCollection.count_documents(
+            {'name': city.name, 'county': city.county, 'state': city.state})
+        if(replace or (not count)):
+            driver.get("https://www.waterqualitydata.us/data/Result/search?within=" + str(within) + "&lat=" + str(city.latitude) + "&long=" + str(city.longitude) +
+                       "&sampleMedia=water&sampleMedia=Water&characteristicName=pH&characteristicName=Total%20dissolved%20solids&characteristicName=Specific%20conductance&startDateLo=" + startDate.strftime("%m-%d-%Y") + "&startDateHi=" + endDate.strftime("%m-%d-%Y") + "&mimeType=csv&zip=yes&sorted=yes&dataProfile=narrowResult")
+            download_wait(directory, 10, 1)
+            upload = readFile(city, directory, within, startDate, endDate, driver)
+            if(upload):
+                uploadToMongoDB(city)
+        print(datetime.now() - startTime)
+
+
+# get lists of cities and set parameters for running the code
+startTime = datetime.now()
+citiesFile = os.getcwd() + "/simplemaps_uscities_basicv1/uscities.csv"
+cities = getCities(citiesFile, "CA")
+#city1 = city(34.4358295, -119.82763890000001, 'Goleta', 'Santa Barbara', 'CA')
+#city2 = city(33.9984235, -118.41173615, 'Los Angeles', 'Los Angeles', 'CA')
+#cities = [city1, city2]
+startDate = datetime.now() - timedelta(days=365)
+endDate = datetime.now()
+replace = True
+
+# connect to MongoDB
+client = MongoClient(
+    'mongodb+srv://guestuser:RktBAzVKMFoI1N7c@cluster0-qbnoy.mongodb.net/test?retryWrites=true&w=majority')
```

Comment:
We shouldn't commit this connection string directly to the repository, is there a way we can put this in a separate file that is referenced in the .gitignore so it doesn't get committed to Github?